### PR TITLE
docs: provide more information on scanning Google's GCR

### PR DESCRIPTION
**Pull request by nobletrout**
I found setting up and using the GCR difficult to figure out the exact credentials needed and the expected file format. Part this was confusion caused ISSUE #783 which wasn't what I needed to know. Hopefully this saves someone else a lot of time.

### CODE REVIEW

File: docs/advanced/private-registries/gcr.md [hunk 1]
```diff
@@ -1,7 +1,41 @@
-Trivy uses Google Cloud SDK. You don't need to install `gcloud` command.
+# Requirements
+None, Trivy uses Google Cloud SDK. You don't need to install `gcloud` command.
 
-If you want to use target project's repository, you can settle via `GOOGLE_APPLICATION_CREDENTIAL`.
+# Privileges
+Credential file should have the `roles/storage.objectViewer` permissions.
```
Comment (knqyf263) [inline docs/advanced/private-registries/gcr.md:5]: I suppose the permission is necessary according to the document. Please correct me if I'm wrong.

```suggestion
Credential file must have the `roles/storage.objectViewer` permissions.
```

File: docs/advanced/private-registries/gcr.md [hunk 2]
```diff
@@ -1,7 +1,41 @@
...
 # must set TRIVY_USERNAME empty char
 export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credential.json
 ```
+
+# Testing
+You can test credentials in the following manner (assuming they are in `/tmp` on host machine).
+
+```bash
+docker run -it --rm -v /tmp:/tmp\
+  -v /var/run/docker.sock:/var/run/docker.sock\
```
Comment (knqyf263) [inline docs/advanced/private-registries/gcr.md:38]: Looks like the socket doesn't need to be mounted.

**Review by knqyf263 — COMMENTED** [review]
Thanks for your contribution!

File: docs/advanced/private-registries/gcr.md (hunk 2 above)
Comment (nobletrout) [reply docs/advanced/private-registries/gcr.md:38]: Just tried without, and you appear to be correct.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**